File: Kirameku-backend/app/schemas/comment.py
from datetime import datetime
from pydantic import BaseModel


class GitHubUserOut(BaseModel):
    id: int
    login: str
    avatar: str
    bio: str


class CommentCreate(BaseModel):
    """评论创建入参。

    新代码用 `target_type`（post/chatter/album）+ `target_id`；
    只传 `post_id` 的老调用方仍然可用（服务层会归一成 post 维度）。
    """

    post_id: int | None = None
    target_type: str | None = None
    target_id: int | None = None
    parent_id: int | None = None
    content: str


class CommentOut(BaseModel):
    id: int
    post_id: int | None = None
    target_type: str = "post"
    target_id: int = 0
    parent_id: int | None
    content: str
    likes: int = 0
    status: str
    created_at: datetime
    github_user: GitHubUserOut | None = None
    replies: list["CommentOut"] = []


# ---- 点赞（P5：用户维度去重，真值在 likes 表，目标表 likes 字段为冗余计数）----

class LikeIn(BaseModel):
    """点赞/取消点赞入参：target_type = chatter / comment / post"""

    target_type: str
    target_id: int


class LikeOut(BaseModel):
    liked: bool
    likes: int


class CommentAdminUpdate(BaseModel):
    status: str  # approved / rejected


# 留言板/杂谈
class MessageCreate(BaseModel):
    content: str
    parent_id: int | None = None


class MessageOut(BaseModel):
    id: int
    github_user_id: int | None
    parent_id: int | None
    content: str
    ip: str = ""
    status: str
    likes: int
    created_at: datetime
    github_user: GitHubUserOut | None = None
    replies: list["MessageOut"] = []


class MessageAdminUpdate(BaseModel):
    status: str

File: Kirameku-backend/admin/src/api/comment.ts
import { http } from "@/utils/http";

/** 评论所属内容（P5 起评论多态：文章 / 说说 / 相册） */
export type CommentTarget = {
  type: "post" | "chatter" | "album" | string;
  id: number;
  /** 标题（说说为内容摘要） */
  title: string | null;
  /** 前台路径，如 /posts/xxx、/moments、/albums */
  url: string | null;
};

export type CommentItem = {
  id: number;
  post_id: number | null;
  target_type: string;
  target_id: number;
  parent_id: number | null;
  content: string;
  ip: string;
  likes?: number;
  status: string;
  created_at: string;
  /** 后台列表才带：所属内容信息 */
  target?: CommentTarget;
  github_user: {
    id: number;
    login: string;
    avatar: string;
    bio: string;
  } | null;
  replies?: CommentItem[];
};

/** 获取文章评论列表（前台用） */
export const getPostComments = (postId: number) => {
  return http.request<CommentItem[]>("get", `/api/comments/post/${postId}`);
};

/** 管理-获取评论列表（可按状态 + 所属内容类型过滤） */
export const getAdminComments = (params?: {
  status?: string;
  target_type?: string;
  page?: number;
  size?: number;
}) => {
  return http.request<CommentItem[]>("get", "/api/comments/admin", { params });
};

/** 更新评论状态 */
export const updateCommentStatus = (commentId: number, status: string) => {
  return http.request("put", `/api/comments/${commentId}/status`, {
    data: { status }
  });
};

/** 删除评论 */
export const deleteComment = (commentId: number) => {
  return http.request<{ ok: boolean }>("delete", `/api/comments/${commentId}`);
};
